diff --git a/hci_ldisc.c.orig b/hci_ldisc.c
index 637c5b8..49d735d 100644
--- a/drivers/bluetooth/hci_ldisc.c
+++ b/drivers/bluetooth/hci_ldisc.c
@@ -801,8 +801,8 @@ static int hci_uart_tty_ioctl(struct tty_struct *tty, struct file *file,
  * We don't provide read/write/poll interface for user space.
  */
 static ssize_t hci_uart_tty_read(struct tty_struct *tty, struct file *file,
-				 unsigned char *buf, size_t nr,
-				 void **cookie, unsigned long offset)
+                 unsigned char __user *buf, size_t nr,
+                 void **cookie, unsigned long offset)
 {
 	return 0;
 }
